cis194/week2/stevemao/LogAnalysis.hs

```diff
@@ -0,0 +1,18 @@
+{-# OPTIONS_GHC -Wall #-}
+module LogAnalysis where
+import Log
+import Text.Read
+
+parseInt:: String -> Int
+parseInt a = maybe 0 id (readMaybe a :: Maybe Int)
+
+-- "I 29 la la la"
+parseMessage :: String -> LogMessage
+parseMessage a = case words a of
+  ("I":x:y:_) -> LogMessage Info (parseInt x) y
+  ("W":x:y:_) -> LogMessage Warning (parseInt x) y
+  ("E":x:y:z:_) -> LogMessage (Error . parseInt $ x) (parseInt y) z
+  _ -> Unknown a
```

Comment:
When readMaybe comes back with Nothing, we should have Unknown messageType.

Reply:
I'll fix it